tests/runtime/watchpoint

```diff
@@ -32,7 +32,8 @@ REQUIRES_FEATURE signal
 
 NAME many_function_probes
 RUN {{BPFTRACE}} -e 'watchpoint:increment+arg0:4:w { printf("hit!\n") }' -c ./testprogs/watchpoint_func_many_probes
-EXPECT Failed to attach watchpoint probe. You are out of watchpoint registers.
+EXPECT ERROR: No more HW registers left
+EXPECT ERROR: Unable to attach probe: watchpoint:./testprogs/watchpoint_func_many_probes:increment:4:w. If this is expected set the 'missing_probes' config variable to 'warn'.
```

Comment:
This is a bit tricky. IIUC, we will not fail in this situation, will we? So issuing an error and the hint is a bit strange.

Reply:
Yeah, this one was weird. Maybe we should just change the behavior to fail as normal if there are no more hardware registers left (and warn if that what the user specified in the config). WDYT?

Reply:
That sounds very much reasonable to me but I assume that there was a reason why it was done this way. Would be good to find it out.

Reply:
@viktormalik was added here: https://github.com/bpftrace/bpftrace/commit/be64c5ba2c592f509befca11fa341b59340659df

I assume we're not aborting if there are no more hardware registers because it's happening in the middle of bpftrace execution (as opposed to at attach time) and perhaps that's a bad experience. It still could potentially cause odd behavior (lost data) in the script so erroring/aborting seems reasonable.

Reply:
I agree that erroring out makes sense but I'd like to get @danobi's opinion.

Reply:
Yeah erroring out makes sense to me